### PR TITLE
1436790/1438462: Part 32: 20 files with MSDN link fixes.

### PR DESCRIPTION
**Pull request by DennisLee-DennisLee**
Contributes to PR #[2037](https://github.com/dotnet/docs/issues/2037) and applies to the following work items:
- [User Story 1436790: Content Update: update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1436790)
- [Task 1438462: Update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1438462)

### CODE REVIEW

File: docs/framework/wcf/guidelines-and-best-practices.md [hunk 1]
```diff
@@ -40,6 +40,6 @@ This section contains topics that provide guidelines for creating Windows Commun
   
 ## See also
 - [What Is Windows Communication Foundation](../../../docs/framework/wcf/whats-wcf.md)
-- [Windows Communication Foundation Samples](https://msdn.microsoft.com/library/8ec9d192-5d81-4f64-bfd3-90c5e5858c91)
+- [Windows Communication Foundation (WCF) samples documentation](./samples/index.md)
```
Comment (mairaw) [inline docs/framework/wcf/guidelines-and-best-practices.md]:
```suggestion
- [Windows Communication Foundation (WCF) samples](./samples/index.md)
```

File: docs/framework/wcf/how-to-implement-an-asynchronous-service-operation.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ dev_langs:
...
+In Windows Communication Foundation (WCF) applications, a service operation can be implemented asynchronously or synchronously without dictating to the client how to call it. For example, asynchronous service operations can be calling synchronously, and synchronous service operations can be called asynchronously. For an example that shows how to call an operation asynchronously in a client application, see [How to: Call Service Operations Asynchronously](../../../docs/framework/wcf/feature-details/how-to-call-wcf-service-operations-asynchronously.md). For more information about synchronous and asynchronous operations, see [Designing Service Contracts](../../../docs/framework/wcf/designing-service-contracts.md) and [Synchronous and Asynchronous Operations](../../../docs/framework/wcf/synchronous-and-asynchronous-operations.md). This topic describes the basic structure of an asynchronous service operation, the code is not complete. For a complete example of both the service and client sides see [Asynchronous](https://docs.microsoft.com/previous-versions/dotnet/netframework-4.0/ms751505(v=vs.100)).
```
Comment (mairaw) [inline docs/framework/wcf/how-to-implement-an-asynchronous-service-operation.md]:
```suggestion
In Windows Communication Foundation (WCF) applications, a service operation can be implemented asynchronously or synchronously without dictating to the client how to call it. For example, asynchronous service operations can be calling synchronously, and synchronous service operations can be called asynchronously. For an example that shows how to call an operation asynchronously in a client application, see [How to: Call Service Operations Asynchronously](../../../docs/framework/wcf/feature-details/how-to-call-wcf-service-operations-asynchronously.md). For more information about synchronous and asynchronous operations, see [Designing Service Contracts](../../../docs/framework/wcf/designing-service-contracts.md) and [Synchronous and Asynchronous Operations](../../../docs/framework/wcf/synchronous-and-asynchronous-operations.md). This topic describes the basic structure of an asynchronous service operation, the code is not complete. For a complete example of both the service and client sides, see [Asynchronous](https://docs.microsoft.com/previous-versions/dotnet/netframework-4.0/ms751505(v=vs.100)).  
```

File: docs/framework/wcf/privacy-information.md [hunk 3]
```diff
@@ -396,5 +396,5 @@ Microsoft is committed to protecting end-users' privacy. When you build an appli
  The Web Services Description Language (WSDL) contains a definition of the port. Each port has an endpoint address and a binding that represents the services used by the application. Exposing WSDL can be turned off using configuration. No information is retained on the machine.  
   
 ## See also
-- [Windows Communication Foundation](https://msdn.microsoft.com/library/fd327ade-0260-4c40-adbe-b74645ba3277)
+- [Developing Service-Oriented Applications with WCF](index.md)
```
Comment (mairaw) [inline docs/framework/wcf/privacy-information.md]: You can keep the link text here that's used in the title metadata

```suggestion
- [Windows Communication Foundation](index.md)
```

File: docs/framework/wcf/getting-started-tutorial.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ The topics contained in this section are intended to give you quick exposure to
 
 The topics in this section assume you are using Visual Studio as the development environment. If you are using another development environment, ignore the Visual Studio-specific instructions.
 
-For sample applications that can be downloaded to your hard disk and run, see the topics in [Windows Communication Foundation Samples](https://msdn.microsoft.com/library/8ec9d192-5d81-4f64-bfd3-90c5e5858c91). For this topic, see, in particular, the [Getting Started](../../../docs/framework/wcf/samples/getting-started-sample.md).
+For sample applications that can be downloaded to your hard disk and run, see the topics in [Windows Communication Foundation (WCF) samples documentation](./samples/index.md). For this topic, see, in particular, the [Getting Started](../../../docs/framework/wcf/samples/getting-started-sample.md).
```
Comment (mairaw) [inline docs/framework/wcf/getting-started-tutorial.md]:
```suggestion
For sample applications that can be downloaded to your hard disk and run, see the topics in [Windows Communication Foundation (WCF) samples](./samples/index.md). For this topic, see, in particular, the [Getting Started](../../../docs/framework/wcf/samples/getting-started-sample.md).
```

File: docs/framework/wcf/samples/ajax-service-using-complex-types-sample.md [hunk 5]
```diff
@@ -6,7 +6,7 @@ ms.assetid: 88242b99-4811-4cbe-8201-52ddf48fb174
 # AJAX Service Using Complex Types Sample
 This sample demonstrates how to use Windows Communication Foundation (WCF) to create an ASP.NET Asynchronous JavaScript and XML (AJAX) service that creates instances of complex types and sends them between service and client as JavaScript Object Notation (JSON). You can access an AJAX service by using JavaScript code from a Web browser client. This sample builds on the [Basic AJAX Service](../../../../docs/framework/wcf/samples/basic-ajax-service.md) sample.  
   
- AJAX support in WCF is optimized for use with ASP.NET AJAX through the <xref:System.Web.UI.ScriptManager> control. For an example of using WCF with ASP.NET AJAX, see the [AJAX Samples](https://msdn.microsoft.com/library/f3fa45b3-44d5-4926-8cc4-a13c30a3bf3e).  
+ AJAX support in WCF is optimized for use with ASP.NET AJAX through the <xref:System.Web.UI.ScriptManager> control. For an example of using WCF with ASP.NET AJAX, see the [AJAX Samples](https://docs.microsoft.com/previous-versions/dotnet/netframework-3.5/bb410780(v=vs.90)).
```
Comment (mairaw) [inline docs/framework/wcf/samples/ajax-service-using-complex-types-sample.md]:
```suggestion
 AJAX support in WCF is optimized for use with ASP.NET AJAX through the <xref:System.Web.UI.ScriptManager> control. For an example of using WCF with ASP.NET AJAX, see the [AJAX Samples](ajax.md).  
```

File: docs/framework/wcf/samples/ajax-service-with-json-and-xml-sample.md [hunk 6]
```diff
@@ -6,7 +6,7 @@ ms.assetid: 8ea5860d-0c42-4ae9-941a-e07efdd8e29c
...
   
- Unlike the other AJAX samples, this sample does not use ASP.NET AJAX and the <xref:System.Web.UI.ScriptManager> control. With some additional configuration, WCF AJAX services can be accessed from any HTML page through JavaScript, and this scenario is shown here. For an example of using WCF with ASP.NET AJAX, see [AJAX Samples](https://msdn.microsoft.com/library/f3fa45b3-44d5-4926-8cc4-a13c30a3bf3e).
+ Unlike the other AJAX samples, this sample does not use ASP.NET AJAX and the <xref:System.Web.UI.ScriptManager> control. With some additional configuration, WCF AJAX services can be accessed from any HTML page through JavaScript, and this scenario is shown here. For an example of using WCF with ASP.NET AJAX, see [AJAX Samples](https://docs.microsoft.com/previous-versions/dotnet/netframework-3.5/bb410780(v=vs.90)).
```
Comment (mairaw) [inline docs/framework/wcf/samples/ajax-service-with-json-and-xml-sample.md]:
```suggestion
 Unlike the other AJAX samples, this sample does not use ASP.NET AJAX and the <xref:System.Web.UI.ScriptManager> control. With some additional configuration, WCF AJAX services can be accessed from any HTML page through JavaScript, and this scenario is shown here. For an example of using WCF with ASP.NET AJAX, see [AJAX Samples](ajax.md).
```

File: docs/framework/wcf/samples/ajax-service-without-configuration.md [hunk 7]
```diff
@@ -6,7 +6,7 @@ ms.assetid: e6db7acd-5679-45d4-b98a-8449c6873838
 # AJAX Service Without Configuration
 This sample demonstrates how to use Windows Communication Foundation (WCF) to create a basic ASP.NET Asynchronous JavaScript and XML (AJAX) service (a service that you can access by using JavaScript code from a Web browser client) without using any configuration settings. The service uses special syntax in the .svc file to automatically enable an AJAX endpoint.  
   
- AJAX support in WCF is optimized for use with ASP.NET AJAX through the `ScriptManager` control. For an example of using WCF with ASP.NET AJAX, see the [Ajax Samples](https://msdn.microsoft.com/library/f3fa45b3-44d5-4926-8cc4-a13c30a3bf3e).  
+ AJAX support in WCF is optimized for use with ASP.NET AJAX through the `ScriptManager` control. For an example of using WCF with ASP.NET AJAX, see the [Ajax Samples](https://docs.microsoft.com/previous-versions/dotnet/netframework-3.5/bb410780(v=vs.90)).
```
Comment (mairaw) [inline docs/framework/wcf/samples/ajax-service-without-configuration.md]:
```suggestion
 AJAX support in WCF is optimized for use with ASP.NET AJAX through the `ScriptManager` control. For an example of using WCF with ASP.NET AJAX, see the [AJAX Samples](ajax).  
```.md

**Review by mairaw — CHANGES_REQUESTED** [review]
Looks great @DennisLee-DennisLee. A few comments.

File: docs/framework/wcf/samples/ajax-service-without-configuration.md (hunk 7 above)
Comment (DennisLee-DennisLee) [reply docs/framework/wcf/samples/ajax-service-without-configuration.md]: The suggested code change is not quite right - I'll fix this one manually.

Comment (mairaw) [reply docs/framework/wcf/samples/ajax-service-without-configuration.md]: Ah sorry @DennisLee-DennisLee. Not sure what happened here.

**Review by mairaw — APPROVED** [review]
Thank you @DennisLee-DennisLee!

**Pull request merged**